shotover-proxy/src/message/mod.rs

```diff
@@ -81,6 +103,7 @@ impl Message {
     /// Providing both the raw bytes and Frame results in better performance if they are both already available.
     pub fn from_bytes_and_frame(bytes: Bytes, frame: Frame) -> Self {
```

Comment:
I think we need to take a ProtocolType here in order to properly set codec_state

Reply:
If you're creating a new message from a frame you can assume that it's not compressed and we can use Compression::None as a default. The Cassandra protocol allows for messages to be compressed/uncompressed, even if compression was agreed on at startup.

Reply:
You are right, we can safely assume that :+1: